rangerequestcheck.php

```diff
@@ -0,0 +1,39 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Performs a range request test
+ *
+ * @package    tool_heartbeat
+ * @copyright  2022
+ * @author     Brendan Heywood <brendan@catalyst-au.net>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+define('NO_MOODLE_COOKIES', true);
+require(__DIR__ . '/../../../config.php');
+require_once(__DIR__.'/nagios.php');
+
+header("Content-Type: text/plain");
+
+// Make sure varnish doesn't cache this. But it still might so go check it!
+header('Pragma: no-cache');
+header('Cache-Control: private, no-cache, no-store, max-age=0, must-revalidate, proxy-revalidate');
+header('Expires: Tue, 04 Sep 2012 05:32:29 GMT');
+
+$check = new \tool_heartbeat\check\rangerequestcheck();
```

Comment:
Will this explode on legacy? No API is no bueno

Reply:
Yes but it's a new endpoint so I don't care about backwards compat, nothing will be calling it